docker-test.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+
+# Strict, and loud failure
+set -euo pipefail
+trap 'rc=$?;set +ex;if [[ $rc -ne 0 ]];then trap - ERR EXIT;echo 1>&2;echo "*** fail *** : code $rc : $DIR/$SCRIPT $ARGS" 1>&2;echo 1>&2;exit $rc;fi' ERR EXIT
+ARGS="$*"
+DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+SCRIPT="$(basename "${BASH_SOURCE[0]}")"
+
+
+if [ "${1:-}" = -h -o "${1:-}" = --help ] ; then
```

Comment:
Wow I'm having a real hard time parsing this. I'm guessing that bash is interpreting `-o` as an OR. Would this be equivalent?
```suggestion
if [ "${1:-}" = -h ] || [ "${1:-}" = --help ] ; then
```

Reply:
My old-school colors are showing! ;-) I never keep all this straight between what bash does internally and when it uses the `[` command (`/bin/test`).  I prefer the direction you're moving it -- it may need double square brackets.  I'll clean it up.

Reply:
Done, with extras that bash `[[` needs in order to do simple matching.